File: src/main/java/com/koddy/server/auth/infrastructure/social/kakao/KakaoOAuthProperties.java
package com.koddy.server.auth.infrastructure.social.kakao;

import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.Set;

@ConfigurationProperties("oauth2.kakao")
public record KakaoOAuthProperties(
        String grantType,
        String clientId,
        String clientSecret,
        String redirectUri,
        Set<String> scope,
        String authUrl,
        String tokenUrl,
        String userInfoUrl
) {
}

File: src/main/kotlin/com/koddy/server/auth/infrastructure/social/kakao/KakaoOAuthProperties.kt
package com.koddy.server.auth.infrastructure.social.kakao

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "oauth2.kakao")
data class KakaoOAuthProperties(
    val grantType: String,
    val clientId: String,
    val clientSecret: String,
    val redirectUri: String,
    val scope: Set<String> = emptySet(),
    val authUrl: String,
    val tokenUrl: String,
    val userInfoUrl: String,
)
